### ISSUE TITLE
statistics "failed" status and more metrics

### ISSUE THREAD

**Issue by highestpeak** [langgenius/dify#6524]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing? Tell me about your story.

**Objective:**

I am configuring a workflow to provide a service that evaluates customer messages for spam. The workflow will:

1. Accept a list of customer messages, each separated by a newline (\n). eg: 
  - "I wanna lookup my order\nHow can i check my orders?\n"
3. Use an LLM to test whether the whole message is spam.
5. Output a JSON result in the format: {"score": "", "reason": ""}.
  - "{\n  \"score\": 2,\n  \"reason\": \"Messages inquire about order lookup and contain minimal content.\"\n}"

**Current Status:**

- The workflow generally works well.
- Occasionally, the LLM outputs multiple scores for each line of the user's message.
  - "{\n  \"messages\": [\n    {\n      \"score\": 0,\n      \"reason\": \"Customer inquiring about order status.\"\n    },\n    {\n      \"score\": 0,\n      \"reason\": \"Customer asking how to check orders.\"\n    },\n    {\n      \"score\": 0,\n      \"reason\": \"Customer providing email and order number.\"\n    },\n    {\n      \"score\": 7,\n      \"reason\": \"Message is meaningless and shows no interest in store.\"\n    }\n  ]\n}"

**Challenges:**

- LLM outputs are inherently uncertain, and we should not rely solely on them.
- While I can check results or errors using code blocks, a dashboard to show mistake frequency would be more efficient.
The logs tab is difficult to use for this purpose.

**Request:**

I am looking for a solution to monitor and display the frequency of these mistakes in a user-friendly dashboard.

### 3. Can you help us with this feature?

- [ ] I am interested in contributing to this feature.

**Comment by alterxyz** [comment]
To improve your spam evaluation workflow:

1. Use an Iteration node to process each message separately:
   - Refer to: <https://docs.dify.ai/guides/workflow/node/iteration>
   - This ensures individual evaluation of each input.
   - The Iteration node handles input and output arrays.

2. Within the Iteration node, implement a Question Classifier:
   - Refer to: <https://docs.dify.ai/guides/workflow/node/question-classifier>
   - In simple terms, a Question Classifier is a pre-prompted, LLM-based single-choice question maker.

Hope this helps!
